xacro/xmlutils.py

```diff
@@ -118,46 +116,49 @@ def check_attrs(tag, required, optional):
     result = reqd_attrs(tag, required)
     result.extend(opt_attrs(tag, optional))
     allowed = required + optional
-    extra = [a for a in tag.attributes.keys() if a not in allowed and not a.startswith("xmlns:")]
+    extra = [
+        a for a in tag.attributes.keys() if a not in allowed and not a.startswith('xmlns:')]
     if extra:
-        warning("%s: unknown attribute(s): %s" % (tag.nodeName, ', '.join(extra)))
+        warning('%s: unknown attribute(s): %s' %
+                (tag.nodeName, ', '.join(extra)))
     return result
 
 
 # Better pretty printing of xml
-# Taken from http://ronrothman.com/public/leftbraned/xml-dom-minidom-toprettyxml-and-silly-whitespace/
-def fixed_writexml(self, writer, indent="", addindent="", newl=""):
+# Taken from
+# http://ronrothman.com/public/leftbraned/xml-dom-minidom-toprettyxml-and-silly-whitespace/
+def fixed_writexml(self, writer, indent='', addindent='', newl=''):
     # indent = current indentation
     # addindent = indentation to add to higher levels
     # newl = newline string
-    writer.write(indent + "<" + self.tagName)
+    writer.write(indent + '<' + self.tagName)
 
     attrs = self._get_attributes()
     a_names = list(attrs.keys())
     a_names.sort()
 
     for a_name in a_names:
-        writer.write(" %s=\"" % a_name)
+        writer.write(' %s=\"' % a_name)
```

Comment:
We have to be careful here.  I believe that we should get rid of the escape slash here.